app/gui2/src/util/shortcuts.ts

```diff
@@ -0,0 +1,400 @@
+    return (event) => {
+      const eventModifierFlags = modifierFlagsForEvent(event)
+      const keybinds = mouseShortcuts[event.buttons as PointerButtonFlags]?.[eventModifierFlags]
+      let handle = handlers[DefaultHandler]
+      if (keybinds != null) {
+        for (const bindingName in handlers) {
+          if (keybinds?.has(bindingName as BindingName)) {
+            handle = handlers[bindingName as BindingName]
+            break
+          }
+        }
+      }
+      if (handle == null) {
+        return false
+      }
+      if (stopImmediatePropagationOnMatch) {
+        event.stopImmediatePropagation()
+      }
+      if (preventDefaultOnMatch) {
+        event.preventDefault()
+      }
+      handle(event)
+      return true
+    }
+  }
+
+  return { keyboardHandler, mouseHandler }
+}
+
+/** A type predicate that narrows the potential child of the array. */
+function includesPredicate<T extends U, U>(array: readonly T[]) {
+  const array_: readonly unknown[] = array
+  return (element: unknown): element is T => array_.includes(element)
+}
+
+const isModifier = includesPredicate(allModifiers)
+const isPointer = includesPredicate(allPointers)
+
+function decomposeKeybindString(string: string): ModifierStringDecomposition {
+  const parts = string
+    .trim()
+    .split(/[\s+]+/)
+    .map((part) => normalizedKeyboardSegmentLookup[part] ?? part)
+  const modifiers = parts.filter(isModifier)
+  const key = parts.find((part) => !isModifier(part))
+  return {
+    // If the key is blank, assume it was a `+` that was removed by the splitting regex.
+    key: key === '' ? '+' : key ?? '',
+    modifiers,
+  }
+}
+
+function parseKeybindString(string: string): Keybind | Mousebind {
+  const decomposed = decomposeKeybindString(string)
+  const normalized =
+    decomposed.modifiers.length === 0
+      ? decomposed.key
+      : `${decomposed.modifiers.join('+')}+${decomposed.key}`
+  if (normalized !== string) {
+    console.warn(`Modifier string '${string}' should be '${normalized}'`)
+  }
+  if (isPointer(decomposed.key)) {
+    return {
+      type: 'mousebind',
+      key: POINTER_BUTTON_FLAG[decomposed.key],
+      modifierFlags: modifierFlagsForModifiers(decomposed.modifiers),
+    }
+  } else {
+    return {
+      type: 'keybind',
+      key: decomposed.key.toLowerCase() as Key_,
+      modifierFlags: modifierFlagsForModifiers(decomposed.modifiers),
+    }
+  }
+}
+
+interface ModifierStringDecomposition {
+  key: string
+  modifiers: Modifier[]
+}
+
+interface Keybind {
+  type: 'keybind'
+  key: Key_
+  modifierFlags: ModifierFlags
+}
+
+interface Mousebind {
+  type: 'mousebind'
+  key: PointerButtonFlags
+  modifierFlags: ModifierFlags
+}
+
+// =================
+// === Constants ===
+// =================
+
+// FIXME [sb]: This should be present in the keybinds too, just like `Mod`.
+/** The key known as the `Delete` key for the current platform. */
+const delete_ = isMacLike ? 'Backspace' : 'Delete'
```

Comment:
I agree, and I think it should be implemented here. Not necessarily as a modifier, but as a key. You could add an extra entry to `allKeys`, and handle it specially during parsing, similarily to '+'. You could potentially add an additional mapping object for use there, similar to modifiers. Just doesn't have to include all keys, only ones needing an override (e.g. `+`, `OsDelete` key).